src/number_theory/legendre_symbol/quadratic_reciprocity.lean

```diff
@@ -199,6 +199,71 @@ end legendre_sym
 
 end legendre
 
+section quadratic_form
+
+/-!
+### Applications to binary quadratic forms
+-/
+
+namespace legendre_sym
+
+/-- The Legendre symbol `legendre_sym p a = 1` if there is a nontrivial solution in `ℤ/pℤ`
+of the equation `x^2 - a*y^2 = 0`. -/
+lemma eq_one_of_sq_sub_mul_sq_eq_zero {p : ℕ} [fact p.prime]
+  {a : ℤ} (ha : (a : zmod p) ≠ 0) {x y : zmod p} (hy : y ≠ 0) (hxy : x ^ 2 - a * y ^ 2 = 0) :
+  legendre_sym p a = 1 :=
+begin
+  apply_fun (* y⁻¹ ^ 2) at hxy,
+  simp only [zero_mul] at hxy,
+  rw [(by ring : (x ^ 2 - ↑a * y ^ 2) * y⁻¹ ^ 2 = (x * y⁻¹) ^ 2 - a * (y * y⁻¹) ^ 2),
+      mul_inv_cancel hy, one_pow, mul_one, sub_eq_zero, pow_two] at hxy,
+  exact (eq_one_iff p ha).mpr ⟨x * y⁻¹, hxy.symm⟩,
+end
+
+/-- The Legendre symbol `legendre_sym p a = 1` if there is a nontrivial solution in `ℤ/pℤ`
+of the equation `x^2 - a*y^2 = 0`. -/
```

Comment:
I was reading docstrings and thinking "wait, we just proved this!". Maybe instead of using the word "nontrivial" twice you could just say "...with x ≠ 0" and "with y ≠ 0"?

Reply:
OK.

Reply:
(Of course, the conditions are equivalent.)